samcli/commands/build/command.py

```diff
@@ -217,13 +218,19 @@ def do_cli(  # pylint: disable=too-many-locals, too-many-statements
             click.secho(msg, fg="yellow")
 
         except (
-            UnsupportedRuntimeException,
-            BuildError,
-            UnsupportedBuilderLibraryVersionError,
-            ContainerBuildNotSupported,
+                UnsupportedRuntimeException,
+                BuildError,
+                BuildInsideContainerError,
+                UnsupportedBuilderLibraryVersionError,
+                ContainerBuildNotSupported,
         ) as ex:
             click.secho("\nBuild Failed", fg="red")
-            raise UserException(str(ex), wrapped_from=ex.__class__.__name__)
+
+            # Some Exceptions have a deeper wrapped exception that needs to be surfaced
+            # from deeper than just one level down.
+            deep_wrap = getattr(ex, "wrapped_from", None)
+            wrapped_from = deep_wrap if deep_wrap else ex.__class__.__name__
```

Comment:
What are other types of value (except class names) which can be used for wrapped_from?

Reply:
technically anything, we can formalize on a specific class maybe if it helps in the future.